cf_xarray/accessor.py

```diff
@@ -608,10 +608,9 @@ def check_results(names, k):
 
     try:
         for name in allnames:
-            extravars = accessor.get_associated_variable_names(name)
-            # we cannot return bounds variables with scalar keys
-            if scalar_key:
-                extravars.pop("bounds")
+            extravars = accessor.get_associated_variable_names(
```

Comment:
shall we instead suppress the warning here using the warnings module. There's probably a way to catch the warning and issue it later after checking `scalar_key`

Reply:
Yes I think that's easy to do but we should raise 2 separate warnings. One for bounds and the other for anything else. That way we still raise a warning if for example lat/lon are missing.

Reply:
https://github.com/xarray-contrib/cf-xarray/blob/48a94858111d302bdb8d72863dc58b6751283a36/cf_xarray/accessor.py#L1266-L1270

Reply:
what if we added the attribute names to warning. Then we could raise the warning `if "bounds" in warning_message and scalar_key`

Reply:
That would help but if we use one warning only we have to:
1. Catch the warning
2. Suppress it if only bounds are in the warning
3. Edit the warning text removing bounds if other variables are missing and issue the new warning

Reply:
Oh I didn't realize (3). 

Hmm.. lets use the kwarg then but rename it to `skip_bounds`?